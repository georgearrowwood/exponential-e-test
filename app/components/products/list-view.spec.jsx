import React from 'react';
import { mount } from 'enzyme';

import ProductsList from './list-view';

const mockProductsList = [
  { id: 1, name: 'test product 1' },
  { id: 2, name: 'test product 2' },
];

let wrapper;
const deleteClickHandlerMock = jest.fn();

beforeAll(() => {
  wrapper = mount(<ProductsList
    products={mockProductsList}
    deleteHandler={deleteClickHandlerMock}
  />);
});

describe('ProductsList View', () => {
  it('Renders products component with properties', () => {
    expect(wrapper).toMatchSnapshot();
    expect(Array.isArray(wrapper.prop('products'))).toEqual(true);
    expect(wrapper.prop('products').length).toEqual(2);
  });

  it('Renders products list items and its titles', () => {
    expect(wrapper.find('tbody').find('tr').length).toEqual(2);
    const firstProduct = wrapper.find('tbody').find('tr').at(0);
    expect(firstProduct.find('td').at(2).text()).toEqual(mockProductsList[0].name);
  });
});
