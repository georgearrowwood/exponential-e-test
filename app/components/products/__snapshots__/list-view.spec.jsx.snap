// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList View Renders products component with properties 1`] = `
<ProductsListView
  deleteHandler={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "name": "test product 1",
      },
      Object {
        "id": 2,
        "name": "test product 2",
      },
    ]
  }
>
  <div>
    <h2
      className="sub-header"
    >
      Products List
    </h2>
    <div
      className="table-responsive"
    >
      <table
        className="table table-striped"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Supplier
            </th>
            <th>
              Product
            </th>
            <th>
              Price
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              1
            </td>
            <td />
            <td>
              test product 1
            </td>
            <td />
          </tr>
          <tr
            key="2"
          >
            <td>
              2
            </td>
            <td />
            <td>
              test product 2
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</ProductsListView>
`;
